src/components/toggle/toggle.st.css

```diff
@@ -0,0 +1,63 @@
+:import {
+    -st-from: "../../style/default-theme/variables.st.css";
+    -st-named:
+        B2,
+        B6,
+        B7,
+        M7,
+        color_MainText_Background;
+}
+:vars {
+	width: 46px;
+	switchSize: 20px;
+}
+.root {
+	-st-states: checked, disabled;
+}
+.background {
+	width: value(width);
+	border: 1px solid value(B7);
+	padding: 1px;
+	border-radius: 100px;
+	background-color: value(color_MainText_Background);
+	cursor: pointer;
+	user-select: none;
+	transition: background-color 200ms linear,
+}
+.input {
+	display: block;
+	opacity: 0;
+	padding: 0;
+	margin: 0;
+	border: 0;
+	width: 0;
+	height: 0;
+}
+
+.switch {
+	height: value(switchSize);
+	width: value(switchSize);
+	background-color: value(B7);
+	border-radius: 100%;
+	display: flex;
+	justify-content: center;
+	align-items: center;
+	transition: all 100ms linear;
+}
+
+.root:checked::switch {
+	transform: translateX(calc(value(width) - 4px - 100%))
```

Comment:
this is `(border + padding) * 2`. Do you want to see this as separate variables?